### ISSUE TITLE
assert type(extras) == dict

### ISSUE THREAD

**Issue by aligoglos** [haotian-liu/yolact_edge#6]
I've tried this script but got error:
`py eval.py --coco_transfer --trained_model=weights/yolact_edge_vid_847_50000.pth --score_threshold=0.3 --top_k=100 --images=./Inputs/:./Results/`
Error:
```
Traceback (most recent call last):
  File "eval.py", line 1399, in <module>
    evaluate(net, dataset)
  File "eval.py", line 868, in evaluate
    evalimages(net, inp, out)
  File "eval.py", line 600, in evalimages
    evalimage(net, path, out_path)
  File "eval.py", line 575, in evalimage
    preds = net(batch)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "D:\Artificial Intelligence\Segmentation\SRC\yolact_edge-master\yolact.py", line 1881, in forward
    assert type(extras) == dict
AssertionError
```

**Comment by aligoglos** [comment]
I manually added `cfg.flow = None` but rised error :
```
Traceback (most recent call last):
  File "eval.py", line 1401, in <module>
    evaluate(net, dataset)
  File "eval.py", line 868, in evaluate
    evalimages(net, inp, out)
  File "eval.py", line 600, in evalimages
    evalimage(net, path, out_path)
  File "eval.py", line 575, in evalimage
    preds = net(batch)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "D:\Artificial Intelligence\Segmentation\SRC\yolact_edge-master\yolact.py", line 1961, in forward
    outs = self.fpn(outs)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\module.py", line 779, in __getattr__
    type(self).__name__, name))
torch.nn.modules.module.ModuleAttributeError: 'Yolact' object has no attribute 'fpn'
```
And added `cfg.fpn = None` but this error occurred!
```
Traceback (most recent call last):
  File "eval.py", line 1401, in <module>
    evaluate(net, dataset)
  File "eval.py", line 868, in evaluate
    evalimages(net, inp, out)
  File "eval.py", line 600, in evalimages
    evalimage(net, path, out_path)
  File "eval.py", line 575, in evalimage
    preds = net(batch)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "D:\Artificial Intelligence\Segmentation\SRC\yolact_edge-master\yolact.py", line 2021, in forward
    p = pred_layer(pred_x)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "D:\Artificial Intelligence\Segmentation\SRC\yolact_edge-master\yolact.py", line 261, in forward
    x = src.upfeature(x)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\container.py", line 117, in forward
    input = module(input)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\conv.py", line 423, in forward
    return self._conv_forward(input, self.weight)
  File "C:\Users\127051\AppData\Local\Programs\Python\Python37\lib\site-packages\torch\nn\modules\conv.py", line 420, in _conv_forward
    self.padding, self.dilation, self.groups)
RuntimeError: Given groups=1, weight of size [256, 256, 3, 3], expected input[1, 512, 69, 69] to have 256 channels, but got 512 channels instead
```

**Comment by haotian-liu** [comment]
For evaluating images, current script seems not fully support some of the models. I will update the evaluation script to support them.

**Comment by aligoglos** [comment]
Thanks.

**Comment by Spritaro** [comment]
Hi, I had a similar problem when I tried to run evaluation for an image using coco models. After some trial and error, I managed to get it working. This might not be a proper solution, but in case anyone is interested, here are the changes I made.

- Add ```'flow': None``` to ```yolact_edge_config``` in data/config.py
- Replace ```preds = net(batch)``` with ```preds = net(batch)['pred_outs']``` in eval.py
- In yolact.py
  - replace ```if cfg.flow.warp_mode == 'none':``` with ```if cfg.flow is not None and cfg.flow.warp_mode == 'none':```
  - add ```if cfg.flow is not None:``` in front of ```self.lat_layer = self.fpn_phase_1.lat_layers[-1]```
  - replace ```if cfg.flow.train_flow:``` with ```if cfg.flow is not None and cfg.flow.train_flow:```
- Remove ```--coco_transfer``` option when running eval.py

**Comment by haotian-liu** [comment]
@aligoglos @Spritaro Fixed in e3a80ee99e396057864c1c603df79496e9089c72.

@Spritaro Thank you for the solution. Note that `coco_transfer` is needed when you are evaluating from a model trained with YOLACT (instead of YolactEdge), or the FPN weights will not be loaded correctly.
